frontend/src/views/HomeView.vue

```diff
@@ -1 +1,68 @@
-<template><h1>Test</h1></template>
+<template>
+    <div class="mb-4">
+      <ElButton
+        key="primary"
+        type="text"
+        @click="showConfirmationDialog(roleId)"
+        style="color: #29517A;"
+      >
+        Delete
+      </ElButton>
+    </div>
+    <ElDialog
+      v-model="dialogVisible"
+      title="Warning"
+      width="500"
+    >
+      <span>Role will be permanently removed. Continue?</span>
+  
+      <template #footer>
+        <div class="dialog-footer">
+          <ElButton @click="dialogVisible = false">Cancel</ElButton>
```

Comment:
All the logic inside the component should be inside script tag (`dialogVisible = false`)